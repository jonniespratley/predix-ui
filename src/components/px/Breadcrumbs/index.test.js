import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';

import PxBreadcrumbs from './';

const selectedRoute = ['North America', 'United States of America', 'California', 'San Ramon'];

const items = [{
  "label": "North America",
  "id": "North America",
  "children": [{
      "label": "United States of America",
      "id": "United States of America",
      "isSelectable": false,
      "children": [{
          "label": "California",
          "id": "California",
          "children": [{
              "label": "Alameda",
              "id": "Alameda"
          }, {
              "label": "Albany",
              "id": "Albany"
          }, {
              "label": "Alhambra",
              "id": "Alhambra"
          }, {
              "label": "Brentwood",
              "id": "Brentwood"
          }, {
              "label": "Burlingame",
              "id": "Burlingame"
          }, {
              "label": "Calistoga",
              "id": "Calistoga"
          }, {
              "label": "Concord",
              "id": "Concord"
          }, {
              "label": "Danville",
              "id": "Danville"
          }, {
              "label": "Emeryville",
              "id": "Emeryville"
          }, {
              "label": "Fresno",
              "id": "Fresno"
          }, {
              "label": "Gilroy",
              "id": "Gilroy"
          }, {
              "label": "Hayward",
              "id": "Hayward"
          }, {
              "label": "Oakland",
              "id": "Oakland"
          }, {
              "label": "San Francisco",
              "id": "San Francisco"
          }, {
              "label": "San Ramon",
              "id": "San Ramon",
              "children": [{
                  "label": "Crow Canyon Gardens",
                  "id": "Crow Canyon Gardens"
              }, {
                  "label": "Old Ranch Park",
                  "id": "Old Ranch Park"
              }, {
                  "label": "Valley View Park",
                  "id": "Valley View Park"
              }],
              "highlighted": false
          }],
          "highlighted": false
      }, {
          "label": "Colorado",
          "id": "Colorado",
          "isSelectable": false
      }, {
          "label": "Delaware",
          "id": "Delaware"
      }, {
          "label": "Florida",
          "id": "Florida"
      }],
      "highlighted": false
  }],
  "highlighted": false
}, {
  "label": "South America",
  "id": "South America"
}];

describe('px-breadcrumbs', () => {
  test('should render', () =>{
    const wrapper = shallow(<PxBreadcrumbs/>);
    console.log(wrapper.debug());
  });

  test('renders correctly', () => {
    const tree = renderer.create(<PxBreadcrumbs items={items} selectedRoute={selectedRoute} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  //expect(wrapper.find('.label')).to.have.length(1);
  //expect(wrapper.find('.delta')).to.have.length(1);
  //expect(wrapper.find('.alpha')).to.have.length(1);
  //expect(wrapper.contains(<div className='label'/>)).to.equal(true);
});
