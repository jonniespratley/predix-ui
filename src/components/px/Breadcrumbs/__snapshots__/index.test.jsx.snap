// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`px-breadcrumbs renders correctly 1`] = `
.c1 {
  color: var(--iron-icon-stroke-color,inherit);
  fill: var(--iron-icon-fill-color,none);
  stroke: var(--iron-icon-stroke-color,currentColor);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  vertical-align: middle;
  width: var(--iron-icon-width);
  height: var(--iron-icon-height);
}

.c1 > svg {
  color: var(--iron-icon-stroke-color,inherit);
  fill: var(--iron-icon-fill-color,none);
  stroke: var(--iron-icon-stroke-color,currentColor);
  position: relative;
  vertical-align: middle;
}

.c1.hidden {
  display: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    className="c0"
  >
    <div
      className="breadcrumbTopItem"
      data-index={0}
    >
      <div
        className="c0"
      >
        <span
          className="actionable actionable--action u-ph-"
          onClick={[Function]}
        >
          North America
        </span>
      </div>
    </div>
    <div
      className="breadcrumbTopItem"
      data-index={1}
    >
      <div
        className="c0"
      >
        <i
          className="px-icon px-utl-chevron-right c1"
          color={null}
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        >
          <svg
            dangerouslySetInnerHTML={
              Object {
                "__html": "<g id=\\"px-utl-chevron-right\\"><path stroke-linejoin=\\"round\\" d=\\"M6.2 13.2l5.4-5.5-5.5-5.5\\"/></g>",
              }
            }
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "display": "block",
                "height": 24,
                "pointerEvents": "none",
                "width": 24,
              }
            }
            viewBox="0 0 16 16"
          />
        </i>
        <span
          className="actionable actionable--action u-ph-"
          onClick={[Function]}
        >
          United States of America
        </span>
      </div>
    </div>
    <div
      className="breadcrumbTopItem"
      data-index={2}
    >
      <div
        className="c0"
      >
        <i
          className="px-icon px-utl-chevron-right c1"
          color={null}
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        >
          <svg
            dangerouslySetInnerHTML={
              Object {
                "__html": "<g id=\\"px-utl-chevron-right\\"><path stroke-linejoin=\\"round\\" d=\\"M6.2 13.2l5.4-5.5-5.5-5.5\\"/></g>",
              }
            }
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "display": "block",
                "height": 24,
                "pointerEvents": "none",
                "width": 24,
              }
            }
            viewBox="0 0 16 16"
          />
        </i>
        <span
          className="actionable actionable--action u-ph-"
          onClick={[Function]}
        >
          California
        </span>
      </div>
    </div>
    <div
      className="breadcrumbTopItem"
      data-index={3}
    >
      <div
        className="c0"
      >
        <i
          className="px-icon px-utl-chevron-right c1"
          color={null}
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        >
          <svg
            dangerouslySetInnerHTML={
              Object {
                "__html": "<g id=\\"px-utl-chevron-right\\"><path stroke-linejoin=\\"round\\" d=\\"M6.2 13.2l5.4-5.5-5.5-5.5\\"/></g>",
              }
            }
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "display": "block",
                "height": 24,
                "pointerEvents": "none",
                "width": 24,
              }
            }
            viewBox="0 0 16 16"
          />
        </i>
        <span
          className="actionable actionable--action u-ph-"
          onClick={[Function]}
        >
          San Ramon
        </span>
      </div>
    </div>
  </div>
</div>
`;
